### DISCUSSION TITLE
Autocomplete org alias CLI parameter

### DISCUSSION THREAD

**Discussion by JamieBriersBullhorn** [forcedotcom/cli#2828] [Ideas]
It would be a really nice QoL improvement, if the CLI could autocomplete org usernames/alias.  

I had managed to achieve this in powershell, by editing `%LOCALAPPDATA%\sf\autocomplete\functions\powershell\sf.ps1`, however it got overridden on upgrade when I refreshed the autocomplete cache.

i.e. I would type in `sf project deploy start -o adm` and it would autocomplete to `sf project deploy start -o adminUsername`

**Comment by cristiand391** [comment]
Hey, yeah this is something I wanted to do too but we haven't decided on it yet.
See issue in plugin-autocomplete: https://github.com/oclif/plugin-autocomplete/issues/620

`sf` uses the [oclif framework](https://oclif.io/) (which we also maintain), shell completion is done in oclif's plugin-autocomplete so what we need is a salesforce-agnostic way to generate completions candidates at runtime. 
If people is interested in this feature please upvote the post.

**Comment by cristiand391** [comment]
Also, if you are on zsh + fzf, I wrote an sf plugin to do this:
https://github.com/forcedotcom/cli/discussions/2569
